lib/pagerduty/maintenance_client.rb

```diff
@@ -6,30 +6,32 @@ module PagerDuty
   class MaintenanceClient < PagerDuty::Service
     configuration PagerDuty::Configuration
 
-    def get_all
+    def get_all(options = {})
       return [] if PagerDuty::Configuration.service_ids.blank? # require whitelisted services
 
-      raw_mws = get_all_raw
+      raw_mws = get_all_raw(options)
       convert(raw_mws)
     end
 
     private
 
-    def get_all_raw
-      resp = get_raw
+    def get_all_raw(options = {})
+      resp = get_raw(options)
       windows = resp['maintenance_windows']
       while resp['more']
         offset = resp['offset'] + resp['limit']
-        resp = get_raw(offset = offset)
+        resp = get_raw(options.merge('offset' => offset))
         windows += resp['maintenance_windows']
       end
       windows
     end
 
-    def get_raw(offset = 0)
```

Comment:
Were there any instances of this being called where `offset` was being overridden to something other than 0 that will now need to be updated to `{offset: 0}`?

Reply:
We're only overriding the offset in the loop where we're reading through everything by incrementing the offset. It's seems to be isolated since `get_raw` is private and `get_all_raw` (also private) was the only place where it was called.